atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java

```diff
@@ -149,13 +154,21 @@ private void scheduleBackgroundThreads() {
                 executorService = PTExecutors
                         .newScheduledThreadPoolExecutor(numThreads, new NamedThreadFactory("Targeted Sweep", true));
                 for (int i = 0; i < numThreads; i++) {
-                    executorService.scheduleWithFixedDelay(
-                            () -> sweepNextBatch(ShardAndStrategy.of(getShardAndIncrement(), sweepStrategy)),
-                            1, 5, TimeUnit.SECONDS);
+                    executorService.scheduleWithFixedDelay(this::runOneIteration, 1, 5, TimeUnit.SECONDS);
                 }
             }
         }
 
+        private void runOneIteration() {
+            ShardAndStrategy shardStrategy = ShardAndStrategy.of(getShardAndIncrement(), sweepStrategy);
+            try {
+                sweepNextBatch(shardStrategy);
+            } catch (Throwable th) {
+                log.error("Targeted sweep for {} failed and will be retried later.",
```

Comment:
Are we guaranteed to try the next `shardStrategy` next time?

We could test it: something like:
`when(sweepNextBatch(any()).thenFail().thenReturn()`
followed by verifying that the arguments to both `sweepNextBatch()` calls are the same.

Reply:
Actually this won't do that if I'm not wrong - we will try if we loop around.

I don't have a nice way of doing this offhand apart from having a queue of failed nodes to try again first, though that might be too much complexity for now?

---

Separately: should this be `warn`, since Background Sweep's analogous message (`The background sweep job failed unexpectedly`) is at warn?

Reply:
Discussed offline -- we are retrying the shard when we come back around to it